docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
+3. The user would be prompted to type the given text as fast as they can.
+4. This game made use of the `CompletableFuture` class to create separate thread which handles the countdown timer and user input at the same time.
+5. When the user finishes typing or the time limit is reached, the game would calculate the user's typing speed and accuracy and reward the user accordingly. 
+6. Finally, the `WorkCommand` would update the player profile with the reward earned according to the user's performance in the game.
+
+## MiniGame - Tic Tac Toe
+
+The implementation of the Tic Tac Toe Game is as follows:
+
+1. The game can be invoked by the `ExerciseCommand` class when the user inputs the `exercise` command.
+2. It makes use of the `ResponseManager` for instructions and display the board status.
+3. The user would be prompted to choose player mark and place it when entering the game.
+4. AI player will randomly choose available to place mark once user places the mark.
+5. After each placement, the game would check the status to determine whether continue the game or not.
+6. If the board is full or there are three consecutive marks, the game will announce the winner or say "it's a draw".
+7. Finally, the `ExerciseCommand` would update the player profile with the reward or punishment for health.
+
+## MiniGame - True or False
+
+1. The game can be invoked by the `RestCommand` class when the user inputs the `rest` command.
+2. It makes use of the `ResponseManager` for instructions and display the questions to answer.
+3. The user would be prompted to answer True or False by typing `T` or `F`.
+4. When the user returns his/her answer, the `ResponseManager` will display the correctness of the response.
+5. When the user finishes answering all the questions, `ResponseManager` will output how many questions the user answers correctly.
+6. Finally, the `RestCommand` would update the player profile with the reward or punishment earned according to the game.
+
+## Investment game - Stock
+
+1. The process of buying stock can be invoked by the `StockCommand` class when user inputs `stock`.
+2. It makes use of `ResponseManager` for instructions and output the stock graph and information relating to the stock.
+3. The user would be prompted to input the number of stocks they would like to purchase in integer.
+4. `ResponseManager` will indicate the completion of purchase after user's input.
+5. When the user finishes the purchase, stock they bought will be stored inside `Asset` class.
+6. Input `status` will output the stock the user currently possesses.
+7. The process of selling stock can be invoked by the `SellStockCommand` class when user input `sellstock`.
+8. The system will calculate user's profit gained at the current price of the stock and return the money back to the 
+user based on the stock's current price.
+9. User can keep the stock they purchased inside their asset for as long as they want.
```

Comment:
Maybe consider adding a diagram of what a stock represents, as it has felt quite arbitrary throughout the entire guide.